### PR TITLE
Feat password at connect dialog iss3856

### LINKED ISSUE #3856: [Feature Request] UI indication for password-protected channels
At the moment there seems to be no indication in Mumble's UI for password-protected channels. Thus making it impossible to tell which channels are password-protected and which are not.

Not even the error message `You were denied Enter privileges in XYZ` makes it clear that the channel is password-protected (one could think one simply doesn't have the permission to enter it when instead only the PW is missing).

I think adding a lock-symbol in the channel "bar" would be nice - See this TS3 screenshot for what I mean: http://img.creativemark.co.uk/uploads/images/558/11558/largeImg.png

In terms of implementation these links might be useful:
https://stackoverflow.com/questions/12316309/how-to-set-an-image-for-a-row
https://stackoverflow.com/questions/22093510/display-image-in-qtreeview

### PR DESCRIPTION
**Pull request by RyanThomasTech**
Fixes #3856 

Added a password column to the connect dialog which will render a lock icon if the server is password-enabled.

I considered simply slotting the lock icon into one of the existing columns, but since the issue's concern was about user ease-of-understanding, I was concerned about users interpreting the icon as being a part of a server's name (if placed in the name column) or that the number of users in a server was limited (if placed in the users column).

Furthermore, adding another column leaves potential room for a "notifications" column of sort, where alerts could be placed for cases like #3158

### CODE REVIEW

**Comment by RyanThomasTech** [comment]
I will squash the previous commit in a few minutes.

File: src/mumble/ConnectDialog.cpp [hunk 1]
```diff
@@ -489,7 +490,11 @@ QVariant ServerItem::data(int column, int role) const {
 				case 2:
 					return uiUsers ? QString::fromLatin1("%1/%2 ").arg(uiUsers).arg(uiMaxUsers) : QVariant();
 			}
-		} else if (role == Qt::ToolTipRole) {
+		} else if (role == Qt::DecorationRole) {
+			if (column == 3){
+				if (!qsPassword.isEmpty())	return QIcon(QLatin1String(":/password_lock.svg"));
+			}
+		}else if (role == Qt::ToolTipRole) {
```
Comment (davidebeatrici) [inline src/mumble/ConnectDialog.cpp]:
```cpp
		} else if (role == Qt::DecorationRole) {
			if (column == 3) {
				if (!qsPassword.isEmpty()) {
					return QIcon(QLatin1String(":/password_lock.svg"));
				}
			}
		} else if (role == Qt::ToolTipRole) {
```

**Comment by davidebeatrici** [comment]
The concept seems good to me!

As for the changes (aside from my comment regarding code style): the icon should be added to https://github.com/mumble-voip/mumble-theme.

A version for the dark theme should be added as well.

**Comment by Johni0702** [comment]
This does **not** address #3856 which is about channel passwords, i.e. showing a lock icon next to channels as is currently done for the comment icon. Solving that issue would probably require changes to the protocol since afaik that information is not currently synced to the client (the channel editor windows pulls it out of the ACL).

This instead adds a lock icon next to servers in the server list which have a password associated with them.
I'm not convinced that this is a good idea because it'll practically only show the lock icon next to servers which you yourself have previously entered the password for (so you probably already know that it's password-protected, unless you've forgotten, but why do you even need to know at this point?).
And more importantly (unless I've misread the code), it won't show the lock icon for any other server, regardless of whether they actually have a password or they don't (that info isn't synced to clients either until they first connect), and that to me seems more confusing than it's worth.

**Comment by RyanThomasTech** [comment]
Yikes, thanks for explaining that John. I was totally misunderstanding the issue. I did think it seemed sort of superfluous at first blush, and now I understand that was due to my interpretation.

**Comment by davidebeatrici** [comment]
@Johni0702 Thank you for pointing that out!

**Comment by Krzmbrzl** [comment]
@RyanThomasTech I'm implementing #3856 alongside the necessary backend-changes in #3929

**Pull request closed without merging**